src/ResourceManager/Websites/Commands.Websites/Cmdlets/WebApps/SetAzureWebApp.cs

```diff
@@ -131,10 +138,40 @@ public override void ExecuteCmdlet()
                             AutoSwapSlotName = parameters.Contains("AutoSwapSlotName") ? AutoSwapSlotName : null,
                             NumberOfWorkers = parameters.Contains("NumberOfWorkers") ? NumberOfWorkers : WebApp.SiteConfig.NumberOfWorkers
                         };
+
+                        // Update web app configuration
+                        WebsitesClient.UpdateWebAppConfiguration(ResourceGroupName, location, Name, null, siteConfig, AppSettings.ConvertToStringDictionary(), ConnectionStrings.ConvertToConnectionStringDictionary());
                     }
 
-                    // Update web app configuration
-                    WebsitesClient.UpdateWebAppConfiguration(ResourceGroupName, location, Name, null, siteConfig, AppSettings.ConvertToStringDictionary(), ConnectionStrings.ConvertToConnectionStringDictionary());
+                    if (parameters.Any(p => CmdletHelpers.SiteParameters.Contains(p)))
+                    {
+                        site = new Site
+                        {
+                            Location = location,
+                            ServerFarmId = WebApp.ServerFarmId,
+                            Identity = parameters.Contains("AssignIdentity") ? new ManagedServiceIdentity("SystemAssigned", null, null) : WebApp.Identity,
+                            HttpsOnly = parameters.Contains("HttpsOnly") ? (bool?)HttpsOnly : null
```

Comment:
@panchagnula how is `null` treated on the server? It seems like this feature is either enabled or disabled, so does `null` default to one of those values?